src/ClimaAtmos/Dycore/src/VanillaAtmosDiscretizations_kernels.jl

```diff
@@ -400,41 +484,80 @@ function volumerhs!(::Val{2}, ::Val{N}, ::Val{nmoist}, ::Val{ntrace},
                     rhs::Array, Q, grad, vgeo, gravity, viscosity, D,
                     elems) where {N, nmoist, ntrace}
   DFloat = eltype(Q)
-
-  nvar = _nstate + nmoist + ntrace
-  ngrad = _nstategrad + 3nmoist
+  nvar   = _nstate + nmoist + ntrace
+  ngrad  = _nstategrad + 3nmoist
+  dim    = 2
 
   Nq = N + 1
 
   nelem = size(Q)[end]
 
-  Q = reshape(Q, Nq, Nq, nvar, nelem)
+  Q    = reshape(Q, Nq, Nq, nvar, nelem)
   grad = reshape(grad, Nq, Nq, ngrad, nelem)
-  rhs = reshape(rhs, Nq, Nq, nvar, nelem)
+  rhs  = reshape(rhs, Nq, Nq, nvar, nelem)
   vgeo = reshape(vgeo, Nq, Nq, _nvgeo, nelem)
 
   s_F = Array{DFloat}(undef, Nq, Nq, _nstate)
   s_G = Array{DFloat}(undef, Nq, Nq, _nstate)
   l_u = Array{DFloat}(undef, Nq, Nq)
   l_v = Array{DFloat}(undef, Nq, Nq)
 
+  q_m = zeros(DFloat, max(3, nmoist))
+    
   @inbounds for e in elems
+
+    delta = Grids.compute_anisotropic_grid_factor(dim, Nq, vgeo, e)
+    delta2 = delta*delta
+    
     for j = 1:Nq, i = 1:Nq
       MJ = vgeo[i, j, _MJ, e]
       ξx, ξy = vgeo[i,j,_ξx,e], vgeo[i,j,_ξy,e]
       ηx, ηy = vgeo[i,j,_ηx,e], vgeo[i,j,_ηy,e]
       y = vgeo[i,j,_y,e]
-
+ 
       U, V = Q[i, j, _U, e], Q[i, j, _V, e]
       ρ, E = Q[i, j, _ρ, e], Q[i, j, _E, e]
-
+        
+      #Moist tracers
+      for m = 1:nmoist
+          s = _nstate + m
+          q_m[m] = Q[i, j, s, e]
+      end
+      (R_m, cp_m, cv_m, gamma_m) = MoistThermodynamics.moist_gas_constants(q_m[1], q_m[2], q_m[3])        
+      gdm1 = R_m/cv_m
       P = gdm1*(E - (U^2 + V^2)/(2*ρ) - ρ*gravity*y)
```

Comment:
Presumably we should be using the function in `MoistThermodynamics` for the pressure?